### DISCUSSION TITLE
New VPS, fixed IP - finally got it running...

### DISCUSSION THREAD

**Discussion by 4044ever** [posativ/isso#990] [Q&A]
Have Ubuntu VPS (thanks to Oracle), installed a LEMP stack via Wordops.net. Also set up isso-comments.xyz (cheapest TLD) and comments.isso-comments.xyz (via wordops). Can reach /index.html

Aim is have isso on an external site, but for now try same site first....

Created a folder `/var/lib/isso/` with `isso.cfg` inside  (some lines deleted)

```
[general]

dbpath = /db/comments.db
name = 
host =
      http://isso-comments.xyz
      https://isso-comments.xyz

max-age = 15m
notify = stdout
reply-notifications=false
log-file =
gravatar = false
gravatar-url = https://www.gravatar.com/avatar/{}?d=identicon&s=55
latest-enabled = false

[admin]
enabled = true
password = 123

[moderation]
enabled = false
approve-if-email-previously-approved = false
purge-after = 30d

[server]
listen = http://localhost:8080
public-endpoint = https://comments.isso-comments.xyz
reload = false
profile = false
trusted-proxies =
samesite =
```

`isso.cfg` is the only file in the `/isso/` folder

```
docker container ls -a
CONTAINER ID   IMAGE                               COMMAND                  CREATED          STATUS                PORTS                      NAMES
011e4cc0f72d   ghcr.io/isso-comments/isso:latest   "/isso/bin/gunicorn …"   15 minutes ago   Up 15 minutes         127.0.0.1:8080->8080/tcp   isso
```

Then `nginx` next....

```
#  /etc/nginx/sites-available/comments.isso-comments.xyz
upstream isso {
    server 127.0.0.1:8080;
}

# HTTP server block for Isso
server {
    listen 80;
    server_name comments.isso-comments.xyz;

    location / {
        proxy_pass http://isso;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
}
```

On the frontend this script worked for me:

```
<script src="https://comments.isso-comments.xyz/js/embed.min.js"></script>
<section id="isso-thread"></section>
```

Looks a little ugly for now, but it works, mostly. On the landing page `index.html` seems the comment disappear. Still need to finde tune a little. So far so good. Special thanks to poe.com providing ChatGPT support.

**Remaining question: How to start more than one container to hast mor comment sites, also for external servers.**

**Comment by ix5** [comment]
> Remaining question: How to start more than one container to hast mor comment sites, also for external servers

You are free to run as many docker containers as your provider will let you. Just take note that they cannot share a port, so the first container will probably live on `8080` and subsequent containers on `8081`, which you will then also have to adjust in the `upstream` blocks of your `nginx` server config.
